src/card.rs

```diff
@@ -45,6 +46,127 @@ use crate::set::{Set, SetCode, SetType};
+            Format::Future => &mut self.future,
+            Format::Pauper => &mut self.pauper,
+            Format::Pioneer => &mut self.pioneer,
+            Format::Penny => &mut self.penny,
+            Format::Duel => &mut self.duel,
+            Format::OldSchool => &mut self.old_school,
+            Format::Historic => &mut self.historic,
+            Format::Gladiator => &mut self.gladiator,
+            Format::Brawl => &mut self.brawl,
+            Format::Premodern => &mut self.premodern,
+            Format::HistoricBrawl => &mut self.historic_brawl,
+            Format::PauperCommander => &mut self.pauper_commander,
+            Format::Alchemy => &mut self.alchemy,
+            Format::Explorer => &mut self.explorer,
+            Format::Predh => &mut self.predh,
+            Format::Oathbreaker => &mut self.oathbreaker,
+            Format::Timeless => &mut self.timeless,
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+/// Scryfall produces multiple sizes of images and image crops for each Card object. Links to these
+/// images are available in each Card objects’ image_uris properties.
+///
+/// Field         | Size       | Format | Example
+///  ---          | ---        | ---    | ---
+/// `png`         | 745 × 1040 | PNG    | [Example Image](https://cards.scryfall.io/png/front/6/d/6da045f8-6278-4c84-9d39-025adf0789c1.png?1562404626)
+/// `border_crop` | 480 × 680  | JPG    | [Example Image](https://cards.scryfall.io/border_crop/front/6/d/6da045f8-6278-4c84-9d39-025adf0789c1.jpg?1562404626)
+/// `art_crop`    | Varies     | JPG    | [Example Image](https://cards.scryfall.io/art_crop/front/6/d/6da045f8-6278-4c84-9d39-025adf0789c1.jpg?1562404626)
+/// `large`       | 672 × 936  | JPG    | [Example Image](https://cards.scryfall.io/large/front/6/d/6da045f8-6278-4c84-9d39-025adf0789c1.jpg?1562404626)
+/// `normal`      | 488 × 680  | JPG    | [Example Image](https://cards.scryfall.io/normal/front/6/d/6da045f8-6278-4c84-9d39-025adf0789c1.jpg?1562404626)
+/// `small`       | 146 × 204  | JPG    | [Example Image](https://cards.scryfall.io/small/front/6/d/6da045f8-6278-4c84-9d39-025adf0789c1.jpg?1562404626)
+pub struct ImageUris {
+    /// A transparent, rounded full card PNG. This is the best image to use for videos or other
+    /// high-quality content.
+    pub png: Url,
+    /// A full card image with the rounded corners and the majority of the border cropped off.
+    /// Designed for dated contexts where rounded images can’t be used.
+    pub border_crop: Url,
+    /// A rectangular crop of the card’s art only. Not guaranteed to be perfect for cards with
+    /// outlier designs or strange frame arrangements
+    pub art_crop: Url,
+    ///  A large full card image
+    pub large: Url,
+    /// A medium-sized full card image
+    pub normal: Url,
+    /// A small full card image. Designed for use as thumbnail or list icon.
+    pub small: Url,
```

Comment:
I tried to find a card that didn't have some of these and I couldn't. Did you find any?

Reply:
that's what I did and it deserialized fine :thinking:

Reply:
Never mind i am stupid. That above is the debug out put from rust not the json. I have though in json was written `image_uris: {}`, which would fail deserialize. But in the json it this field simple missing, so using Option for this is fine.